correcting compilation and execution errors in part-1

pkulkarni4 opened the pull request:
correcting compilation and execution errors in part-1

pkulkarni4 committed:
Update predicate.java

Correcting typo predicate, avoiding compilation error in jshell.

pkulkarni4 committed:
Merge pull request #1 from pkulkarni4/pkulkarni4-jshell-error

Update predicate.java

pkulkarni4 committed:
Update predicate.java

Error while executing the jshell. incompatible types: boolean cannot be converted to java.lang.Integer

pkulkarni4 committed:
Merge pull request #2 from pkulkarni4/pkulkarni4-jshell-execution-error

Update predicate.java

benweidig commented:
Thanks for the errata/PR!

I‘m on vacation right now, but I will take a look as soon as possible.

benweidig committed:
Update predicate.java to match book and correct return type

The lambda variable now matches the book, and the return type uses a "small" boolean.
